### PR TITLE
GL updates

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: Data/Base.rte/Shaders/ScreenBlit.frag [hunk 1]
```diff
@@ -20,5 +20,5 @@ void main() {
 	vec4 guiColor = texture2DAA(rteGUITexture, textureUV);
 	float guiSolid = step(0.00000001, guiColor.r + guiColor.g + guiColor.b);
 	float blendRatio = max(guiColor.a, guiSolid);
-	FragColor = (texture2DAA(rteTexture, textureUV) * (1 - blendRatio)) + guiColor * blendRatio;
+	FragColor = (texture2DAA(rteTexture, textureUV) * (1.0 - blendRatio)) + guiColor * blendRatio;
```
Comment (Causeless) [inline Data/Base.rte/Shaders/ScreenBlit.frag:23]: Why not 1.0F?

File: Data/Base.rte/Shaders/Blit8.frag [hunk 2]
```diff
@@ -6,9 +6,9 @@ in vec2 textureUV;
 out vec4 FragColor;
 
 uniform sampler2D rteTexture;
-uniform sampler1D rtePalette;
+uniform sampler2D rtePalette;
 
 void main() {
 	float colorIndex = texture(rteTexture, vec2(textureUV.x, -textureUV.y)).r;
-	FragColor = texture(rtePalette, colorIndex);
+	FragColor = texture(rtePalette, vec2(colorIndex, 0.f));
```
Comment (Causeless) [inline Data/Base.rte/Shaders/Blit8.frag:13]: We typically format floats always including the 0 and capitalised F, so 0.0F

File: Data/Base.rte/Shaders/Blit8.frag [hunk 3]
```diff
@@ -6,9 +6,9 @@ in vec2 textureUV;
 out vec4 FragColor;
 
 uniform sampler2D rteTexture;
-uniform sampler1D rtePalette;
```
Comment (Causeless) [inline Data/Base.rte/Shaders/Blit8.frag:9]: Why? Are 1d textures unsupported on some platforms or something?

**Review by Causeless — CHANGES_REQUESTED** [review]
Couple comments, but I don't know much about shader programming so feel free to swat em away 🫡

File: Data/Base.rte/Shaders/Blit8.frag (hunk 3 above)
Comment (HeliumAnt) [reply Data/Base.rte/Shaders/Blit8.frag:9]: I'm guessing for gles support? Anything else should have support for tex1d, though I'm guessing driver implementation is probably mostly same as this here.

File: Data/Base.rte/Shaders/ScreenBlit.frag (hunk 1 above)
Comment (HeliumAnt) [reply Data/Base.rte/Shaders/ScreenBlit.frag:23]: glsl implicitly uses float for everything, specfying f is redundant

File: Data/Base.rte/Shaders/Blit8.frag (hunk 3 above)
Comment (janisozaur) [reply Data/Base.rte/Shaders/Blit8.frag:9]: Yes

**Comment by Causeless** [comment]
Does implementing GLES at runtime give us any real benefit? Does that just mean we can query for the best API at startup instead of  (necessarily) providing different binaries, or is there some other reason?

**Review by Causeless — APPROVED** [review]

**Pull request merged**